File: LeetCode/0554. Brick Wall/solution.py
"""
87 / 87 test cases passed.
Runtime: 72  ms
Memory Usage: 18 MB
"""
from itertools import accumulate
from collections import Counter
class Solution:
    def leastBricks(self, wall: List[List[int]]) -> int:
        through = []
        for bricks in wall:
            through.extend(accumulate(bricks[:-1]))
        max_through = Counter(through).most_common()
        return len(wall) - (max_through[0][-1] if len(max_through) else 0)

File: LeetCode/0554. Brick Wall/solution.cpp
/**
 * 87 / 87 test cases passed.
 * Runtime: 48 ms
 * Memory Usage: 21.1 MB 
 */
class Solution {
public:
    int leastBricks(vector<vector<int>>& wall) {
        unordered_map<int, int> store;
        for (auto& bricks: wall) {
            int width = 0;
            for (int i = 0; i < bricks.size() - 1; ++ i) {
                width += bricks[i];
                store[width] += 1;
            }
        }
        int through = 0;
        for (auto& [width, count]: store) {
            through = max(through, count);
        }
        return wall.size() - through;
    }
};
